android/java/org/chromium/chrome/browser/component_updater/BraveComponentUpdater.java

```diff
@@ -0,0 +1,161 @@
+        ThreadUtils.assertOnUiThread();
+        mComponentUpdaterListeners.add(listener);
+    }
+
+    public void removeComponentUpdateEventListener(ComponentUpdaterListener listener) {
+        ThreadUtils.assertOnUiThread();
+        mComponentUpdaterListeners.remove(listener);
+    }
+
+    @CalledByNative
+    protected void componentStateUpdated(int event, String id) {
+        for (ComponentUpdaterListener listener : mComponentUpdaterListeners) {
+            listener.onComponentUpdateEvent(event, id);
+        }
+    }
+
+    /**
+     * Class describing the progress of component download.
+     */
+    public class CrxUpdateItem {
+        // From components/update_client/update_client.h
+        public static final int STATUS_NEW = 0;
+        public static final int STATUS_CHECKING = 1;
+        public static final int STATUS_CAN_UPDATE = 2;
+        public static final int STATUS_DOWNLOADING_DIFF = 3;
+        public static final int STATUS_DOWNLOADING = 4;
+        public static final int STATUS_DOWNLOADED = 5;
+        public static final int STATUS_UPDATING_DIFF = 6;
+        public static final int STATUS_UPDATING = 7;
+        public static final int STATUS_UPDATED = 8;
+        public static final int STATUS_UP_TO_DATE = 9;
+        public static final int STATUS_UPDATE_ERROR = 10;
+        public static final int STATUS_UNINSTALLED = 11;
+        public static final int STATUS_REGISTRATION = 12;
+        public static final int STATUS_RUN = 13;
+        public static final int STATUS_LAST_STATUS = 14;
+
+        public String mId;
+        public long mDownloadedBytes = -1;
+        public long mTotalBytes = -1;
+        public int mState = -1;
+
+        public boolean isInProgress() {
+            return mState == STATUS_DOWNLOADING_DIFF || mState == STATUS_DOWNLOADING
+                    || mState == STATUS_DOWNLOADED || mState == STATUS_UPDATING_DIFF
+                    || mState == STATUS_UPDATING;
+        }
+
+        public void normalizeZeroProgress() {
+            if (mState == STATUS_DOWNLOADING && mTotalBytes == -1 && mDownloadedBytes == -1) {
+                mTotalBytes = 100;
+                mDownloadedBytes = 0;
+            }
+        }
+    }
+
+    public CrxUpdateItem getUpdateState(String id) {
+        String json = BraveComponentUpdaterJni.get().getUpdateState(
+                mNativeBraveComponentUpdaterAndroid, id);
+
+        CrxUpdateItem crxUpdateItem = new CrxUpdateItem();
+        try {
+            JSONObject result = new JSONObject(json);
+            crxUpdateItem.mId = result.getString("id");
+            crxUpdateItem.mDownloadedBytes = (long) result.getDouble("downloaded_bytes");
+            crxUpdateItem.mTotalBytes = (long) result.getDouble("total_bytes");
+            crxUpdateItem.mState = (int) result.getInt("state");
+        } catch (JSONException e) {
+            Log.e(TAG, "getUpdateState JSONException error " + e);
```

Comment:
Let's not include the error in the error message. Better pass the entire stacktrace to the appropriate parameter:
 `Log.e(TAG, "getUpdateState JSONException error", e);`

Reply:
thanks, fixed at https://github.com/brave/brave-core/pull/19086/commits/54818d8dcc99e3f267ebe48dde6da4ba6625545f#diff-b0ad67a3cc1f46d4b3163f2b99f39d57f940215f0c4a0a327a02bea2cf99f35cR147